### DISCUSSION TITLE
Help Understanding What Runs with responseType = arraybuffer

### DISCUSSION THREAD

**Discussion by nhilbelink** [axios/axios#6057] [Q&A]
I proto-typed a GET command using axios. It returns a gzipped report from Apples app store api. At first it returned encoded data that is unreadable(see pic below). Eventually I got the command to work by including responseType = arraybuffer. However, due to restrictions with a 3rd party I have to try and recreate the command outside of axios, but I am back to getting the encoded text. I've tried to parse through the code myself to figure out what commands are being run due to this config setting, but it is a bit over my head.

Any help would be greatly appreciated.

![image](https://github.com/axios/axios/assets/119458024/40068ff8-2470-4aee-b4ed-6e58aec2093b)

**Comment by jasonsaayman** [comment]
`responseType: 'arraybuffer'` tells Axios not to treat the response body as text/JSON. It keeps the response as raw bytes.

For a gzipped report, there are two separate things involved:

1. `Content-Encoding: gzip`
   This is HTTP-level compression. In Node.js, Axios' HTTP adapter normally decompresses this automatically unless `decompress: false` is set.

2. `responseType: 'arraybuffer'`
   This controls how the already-received body is represented to your code. With `arraybuffer`, you get bytes instead of decoded text.

If you are recreating this outside Axios, make sure you are doing both pieces correctly:

- Accept/read the response as binary data, not a string.
- If the server sends `Content-Encoding: gzip`, decompress the bytes with gzip before interpreting the report.

In Node.js that usually means using `zlib.gunzip`/`gunzipSync` or piping through `zlib.createGunzip()`, depending on how you are reading the response.
